PDSim/core/core.py

```diff
@@ -1114,43 +1128,68 @@ def OBJECTIVE_CYCLE(self, Td_Tlumps0, X, epsilon = 0.003, cycle_integrator = 'RK
             ###  -----------------------------------
             ###         The lump temperatures
             ###  -----------------------------------
-                
+    
             self.solvers.lump_eb_history.append([self.Tlumps, self.lumps_resid])
-            
-            if len(self.solvers.lump_eb_history) == 1:
+
+            if len(self.Tlumps) > 1:
                 
```

Comment:
I need a more robust way of handling multi-lump. Probably a Broyden-type of solver.